src/main/java/seedu/organizer/model/task/DateCompleted.java

```diff
@@ -0,0 +1,108 @@
+package seedu.organizer.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.organizer.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+
+//@@author natania
+/**
+ * Represents a Task's dateCompleted in the organizer book.
+ * Guarantees: immutable;
+ */
+public class DateCompleted {
+
+    public static final String MESSAGE_DATECOMPLETED_CONSTRAINTS =
+            "Dates should be in the format YYYY-MM-DD, and it should not be blank";
+
+    /*
+     * The first character must not be a whitespace, otherwise " " (a blank string) becomes a valid input.
+     * Format of string is YYYY-MM-DD.
+     */
+    public static final String DATECOMPLETED_VALIDATION_REGEX = "\\d{4}-(0[1-9]|1[0-2])-(0[1-9]|[12]\\d|3[01])";
+    public static final String TASK_NOTCOMPLETED = "not completed";
+
+    public final LocalDate date;
+
+    /**
+     * Constructs an {@code DateCompleted}.
+     *
+     * @param dateCompleted A valid date.
+     */
+    public DateCompleted(String dateCompleted) {
+        requireNonNull(dateCompleted);
+        checkArgument(isValidDateCompleted(dateCompleted), MESSAGE_DATECOMPLETED_CONSTRAINTS);
+        //temporary fix for xml file bug due to PrioriTask's dependence on the current date
+        if (dateCompleted.equals("current_date")) {
+            this.date = LocalDate.now();
+        } else {
+            //actual code that is run when tests are not running
+            if (dateCompleted.equals(TASK_NOTCOMPLETED)) {
+                this.date = null;
+            } else {
+                this.date = LocalDate.parse(dateCompleted);
+            }
+        }
+    }
+
+    /**
+     * Constructs a DateCompleted based on the current date
+     */
+    public DateCompleted() {
+        LocalDate currentDate = LocalDate.now();
+        requireNonNull(currentDate);
+        this.date = currentDate;
+    }
+
+    /**
+     * Constructs a DateCompleted that is empty
+     */
+    public DateCompleted(boolean completed) {
```

Comment:
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. name for boolean doesn't follow naming convention